kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionMetadataFactory.java

```diff
@@ -692,7 +695,12 @@ private static void validateSchemaAndPartColsCreateOrReplace(
     // New table verify the given schema and partition columns
     ColumnMappingMode mappingMode = ColumnMapping.getColumnMappingMode(tableProperties);
 
-    SchemaUtils.validateSchema(schema, isColumnMappingModeEnabled(mappingMode));
+    SchemaUtils.validateSchema(
+        schema,
+        isColumnMappingModeEnabled(mappingMode),
+        TableFeatures.isPropertiesManuallySupportingTableFeature(
+            tableProperties, ALLOW_COLUMN_DEFAULTS_W_FEATURE),
+        TableConfig.ICEBERG_COMPAT_V3_ENABLED.fromMetadata(tableProperties));
```

Comment:
I think @amogh-jahagirdar merged a PR that changes where we do this validation -- you should be able to just use the protocol for this after rebasing